archunit/src/main/java/com/tngtech/archunit/lang/ArchRule.java

```diff
@@ -59,6 +61,10 @@
  * @see com.tngtech.archunit.library.dependencies.Slices.Transformer
  */
 public interface ArchRule extends CanBeEvaluated, CanOverrideDescription<ArchRule> {
+
+    @PublicAPI(usage = ACCESS)
+    String VERIFY_INPUT_NOT_EMPTY_PROPERTY_NAME = "archRule.verifyInputNotEmpty";
```

Comment:
Maybe we can name the property a little different? Because "input" is a little ambiguous for me. Generally rules work the following way:

```
allImportedClasses -> rule.that(..) -> filteredClasses -> rule.should(..) -> checks filteredClasses
```

So "input" could also refer to `allImportedClasses`, but what we're testing is even more strict (it would fail if `allImportedClasses` is empty but also on `filteredClasses` being empty). Maybe `failOnEmptyCheck`? Or `failOnEmptyShould`? :thinking:

Reply:
> I don't think I would make this public API.

Yes, totally agree. This was just a quick fix for a failing `PublicAPIRules`-rule 😅

Reply:
I like your suggested renaming. I prefer `failOnEmptyShould` over `failOnEmptyCheck` since, for me, with `check` it is not totally clear what check is meant. I renamed it also accordingly

Reply:
Regarding `@PublicAPI`: I probably didn't express this very well :upside_down_face: What I meant is don't make this public at all. You can move it into `SimpleArchRule`, which is a class (and also the real user of this property), and make it private. In the test I would rather use the raw string then :man_shrugging: